src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -16271,5 +16271,166 @@ public void RestrictedType_02([CombinatorialValues("record", "record class", "re
+        {
+            var src1 = """
+                using System;
+
+                [method: Obsolete("")]
+                class Point(int i)
+                {
+                    public int I { get; } = i;
+                }
+                """;
+            var comp = CreateCompilation(src1, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees.Single();
+            var semanticModel = comp.GetSemanticModel(tree);
+            var typeDeclaration = tree.GetRoot().ChildNodes().OfType<TypeDeclarationSyntax>().Single();
+
+            var symbols = semanticModel.GetDeclaredSymbolsForNode(typeDeclaration);
+            Assert.Equal(2, symbols.Length);
+
+            var namedType = symbols.OfType<INamedTypeSymbol>().Single();
+            var primaryConstructor = symbols.OfType<IMethodSymbol>().Single();
+
+            Assert.Same(primaryConstructor, namedType.GetSymbol<SourceMemberContainerTypeSymbol>().PrimaryConstructor.GetPublicSymbol());
+            Assert.Equal(1, primaryConstructor.GetAttributes().Length);
+        }
+
+        [Fact]
+        public void SemanticModel_GetDeclaredSymbols2()
+        {
+            var src1 = """
+                using System;
+
+                [method: Obsolete("")]
+                partial class Point(int i)
+                {
+                    public int I { get; } = i;
+                }
+
+                partial class Point
+                {
+                }
+                """;
+            var comp = CreateCompilation(src1, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees.Single();
+            var semanticModel = comp.GetSemanticModel(tree);
+            var root = tree.GetRoot();
+            var typeDeclaration1 = root.ChildNodes().OfType<TypeDeclarationSyntax>().First();
+
+            var symbols1 = semanticModel.GetDeclaredSymbolsForNode(typeDeclaration1);
+            Assert.Equal(2, symbols1.Length);
+
+            var namedType1 = symbols1.OfType<INamedTypeSymbol>().Single();
+            var primaryConstructor1 = symbols1.OfType<IMethodSymbol>().Single();
+
+            Assert.Same(primaryConstructor1, namedType1.GetSymbol<SourceMemberContainerTypeSymbol>().PrimaryConstructor.GetPublicSymbol());
+            Assert.Equal(1, primaryConstructor1.GetAttributes().Length);
+
+            var typeDeclaration2 = root.ChildNodes().OfType<TypeDeclarationSyntax>().Last();
+            var symbols2 = semanticModel.GetDeclaredSymbolsForNode(typeDeclaration2);
+            Assert.Equal(1, symbols2.Length);
+
+            var namedType2 = symbols2.OfType<INamedTypeSymbol>().Single();
+            Assert.Equal(namedType1, namedType2);
+        }
+
+        [Fact]
+        public void SemanticModel_GetDeclaredSymbols3()
+        {
+            var src1 = """
+                using System;
+
+                [method: Obsolete("")]
```

Comment:
>[method: Obsolete("")] [](http://example.com/codeflow?start=16&length=22)

Consider also testing scenario where the attribute is on the second part and both parts have parameters

Reply:
sounds good.